pkg/ccl/changefeedccl/compression.go

```diff
@@ -55,6 +55,27 @@ func newCompressionCodec(
 	}
 }
 
+// newDecompressionReader returns decompression reader for the specified algorithm
+func newDecompressionReader(algo compressionAlgo, src io.Reader) (io.ReadCloser, error) {
+	switch algo {
+	case sinkCompressionGzip:
+		// since we are using decompression only for reading error response body, we can use default reader
+		return pgzip.NewReader(src)
+	case sinkCompressionZstd:
+		// zstd reader does not implement io.Closer interface, so we need to wrap it
```

Comment:
The zstd reader does implement Close(), which we should call.

It may also be worth caching the readers instead of making a new one every time... but that might be unnecessary for this case.

Reply:
zstd reader implements `func (d *[Decoder] Close()` void
but `io.ReadCloser` requires
`func (io.Closer) Close() error` with error
that's why I had to wrap it

Reply:
Yes, but your wrapper does not call the void Close method, where it should.

Reply:
[Fixed](https://github.com/cockroachdb/cockroach/commit/e8616aa9f2e217cc3622163e633c088b36f4c487)